core/integration/features/fees/0029-FEES-high_volume_maker.feature

```diff
@@ -58,26 +61,102 @@ Feature: Discounts from multiple sources
+      | party  | market id | peak size | minimum visible size | side | pegged reference | volume | offset |
+      | lpprov | ETH/USD   | 5000      | 1000                 | buy  | BID              | 10000  | 1      |
+      | lpprov | ETH/USD   | 5000      | 1000                 | sell | ASK              | 10000  | 1      |
 
     When the parties place the following orders:
-      | party | market id   | side | volume | price | resulting trades | type       | tif     |
-      | aux1  | ETH/USD | buy  | 1      | 990   | 0                | TYPE_LIMIT | TIF_GTC |
-      | aux1  | ETH/USD | buy  | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
-      | aux2  | ETH/USD | sell | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
-      | aux2  | ETH/USD | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party | market id | side | volume | price | resulting trades | type       | tif     |
+      | aux1  | ETH/USD   | buy  | 1      | 990   | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux1  | ETH/USD   | buy  | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux2  | ETH/USD   | sell | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux2  | ETH/USD   | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_GTC |
 
     Then the opening auction period ends for market "ETH/USD"
     And the trading mode should be "TRADING_MODE_CONTINUOUS" for the market "ETH/USD"
 
+  Scenario: When there is `high_volume_market_maker_rebate`, `high_volume_maker_fee` should be taken from the `treasury/buyback_fee` components with value `high_volume_maker_fee = high_volume_factor * trade_value_for_fee_purposes` (0029-FEES-042, 0029-FEES-043, 0029-FEES-044, 0029-FEES-047)
+
+    Given the parties place the following orders:
+      | party   | market id | side | volume | price | resulting trades | type       | tif     |
+      | trader1 | ETH/USD   | sell | 210    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | trader2 | ETH/USD   | sell | 310    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | trader3 | ETH/USD   | sell | 480    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux1    | ETH/USD   | buy  | 1000   | 1000  | 3                | TYPE_LIMIT | TIF_GTC |
+
+    And the network moves ahead "1" epochs
+    Given the parties place the following orders:
+      | party   | market id | side | volume | price | resulting trades | type       | tif     |
+      | trader1 | ETH/USD   | sell | 210    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | trader2 | ETH/USD   | sell | 310    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | trader3 | ETH/USD   | sell | 480    | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux1    | ETH/USD   | buy  | 1000   | 1000  | 3                | TYPE_LIMIT | TIF_GTC |
+
+    And the network moves ahead "1" epochs
```

Comment:
```suggestion
    And the network moves ahead "1" epochs
    Then clear transfer response events
```

At the moment when we check the transfers below there is no guarantee they are the transfers resulting from the trades in this epoch.

Ideally we would check the trade and fees generated directly in a step but we could also just clear the memory before generating the trades.